yt_dlp/extractor/radiofrance.py

```diff
@@ -102,3 +144,327 @@ def _real_extract(self, url):
+                        'timestamp': 'publishedDate',
+                        'thumbnail': ('visual', 'src'),
+                    }))
+
+            next_cursor = profile_response['pagination']['next']
+            if not next_cursor:
+                break
+
+            profile_response = self._download_json(
+                f'https://www.radiofrance.fr/api/v2.1/taxonomy/{profile_id}/documents', profile_id,
+                note=f'Downloading page {page_num}', query={
+                    'relation': 'personality',
+                    'cursor': next_cursor
+                })
+
+
+class RadioFranceProgramScheduleIE(RadioFranceBase):
+    _VALID_URL = rf'''(?x)
+        {RadioFranceBase._VALID_URL_BASE}
+        /(?P<station>{"|".join(RadioFranceBase._STATIONS)})
+        /grille-programmes(?:\?date=(?P<date>[\d-]+))?
+    '''
+
+    _TESTS = [{
+        'url': 'https://www.radiofrance.fr/franceinter/grille-programmes?date=17-02-2023',
+        'info_dict': {
+            'id': 'franceinter-program-20230217',
+            'upload_date': '20230217',
+        },
+        'playlist_count': 25,
+    }, {
+        'url': 'https://www.radiofrance.fr/franceculture/grille-programmes?date=01-02-2023',
+        'info_dict': {
+            'id': 'franceculture-program-20230201',
+            'upload_date': '20230201',
+        },
+        'playlist_count': 25,
+    }, {
+        'url': 'https://www.radiofrance.fr/mouv/grille-programmes?date=19-03-2023',
+        'info_dict': {
+            'id': 'mouv-program-20230319',
+            'upload_date': '20230319',
+        },
+        'playlist_count': 3,
+    }, {
+        'url': 'https://www.radiofrance.fr/francemusique/grille-programmes?date=18-03-2023',
+        'info_dict': {
+            'id': 'francemusique-program-20230318',
+            'upload_date': '20230318',
+        },
+        'playlist_count': 15,
+    }, {
+        'url': 'https://www.radiofrance.fr/franceculture/grille-programmes',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        station, date = self._match_valid_url(url).group('station', 'date')
+
+        api_response = self._download_json(
+            f'https://www.radiofrance.fr/api/v2.1/stations/{station}/programs', f'{station}-program',
+            query={'date': date} if date is not None else {})
+
+        entries = []
+        for entry in api_response['steps']:
+            path = traverse_obj(entry, ('expression', 'path'))
+            if path is None:
+                continue
+
+            entries.append(self.url_result(
+                urljoin(url, f'/{path}'), ie=FranceCultureIE, **traverse_obj(entry, {
+                    'title': ('expression', 'title'),
+                    'thumbnail': ('expression', 'visual', 'src'),
+                    'timestamp': 'startTime',
+                    'series_id': ('concept', 'id'),
+                    'series': ('concept', 'title'),
+                })))
```

Comment:
Not mandatory, but u could move this to a generator instead of init-append loop

Reply:
do we want this to be `url_transparent=True` since most of this metadata is not extracted by `FranceCultureIE`? (e.g. `series_id`, `series`, `timestamp`)